lib/shopify_transporter/exporters/magento/product_exporter.rb

```diff
@@ -20,6 +24,31 @@ def base_products
           result = @client.call(:catalog_product_list, filters: nil).body
           result[:catalog_product_list_response][:store_view][:item] || []
         end
+
+        def product_mappings
+          product_mapping_exporter.write_mappings(@intermediate_file_name)
+
+          @product_mappings ||= {}.tap do |product_mapping_table|
+            CSV.read(@intermediate_file_name).each do |(parent_id, child_id)|
+              product_mapping_table[child_id] = parent_id
+            end
+          end
+        end
+
+        def product_mapping_exporter
+          @product_mapping_exporter ||= ProductMappingExporter.new(@database_adapter)
+        end
+
+        def apply_mappings(product_list)
+          product_list.map do |product|
+            case product[:type]
+            when 'simple'
+              product.merge(parent_id: product_mappings[product[:product_id]])
```

Comment:
add `unless product_mappings[product[:product_id].nil?`

Reply:
if we're adding the condition `unless product_mappings[product[:product_id].nil?`, we should extract that out into a private method
```ruby
when 'simple'
   merge_simple_with_parent(product, product_mappings)
else
   product
end